src/main/java/org/terasology/corerendering/rendering/dag/nodes/ShadowMapNode.java

```diff
@@ -162,18 +164,21 @@ public void process() {
             int numberOfChunksThatAreNotReadyYet = 0;
 
             final Vector3f cameraPosition = shadowMapCamera.getPosition();
+            shadowMapMaterial.setMatrix4("projection", shadowMapCamera.getProjectionMatrix(), true);
 
-            shadowMapCamera.lookThrough();
-
+            Matrix4f modelViewMatrix = new Matrix4f();
+            Matrix4f model = new Matrix4f();
             // FIXME: storing chunksOpaqueShadow or a mechanism for requesting a chunk queue for nodes which calls renderChunks method?
             while (renderQueues.chunksOpaqueShadow.size() > 0) {
                 RenderableChunk chunk = renderQueues.chunksOpaqueShadow.poll();
-
                 if (chunk.hasMesh()) {
-                    final ChunkMesh chunkMesh = chunk.getMesh();
-                    final Vector3f chunkPosition = new Vector3f(chunk.getPosition(new Vector3i()));
-
-                    numberOfRenderedTriangles += chunkMesh.render(OPAQUE, chunkPosition, cameraPosition);
+                    model.setTranslation(
+                            chunk.getPosition().x() * Chunks.SIZE_X - cameraPosition.x(),
+                            chunk.getPosition().y() * Chunks.SIZE_Y - cameraPosition.y(),
+                            chunk.getPosition().z() * Chunks.SIZE_Z - cameraPosition.z());
+                    modelViewMatrix.set(shadowMapCamera.getViewMatrix()).mul(model);
+                    shadowMapMaterial.setMatrix4("modelView", modelViewMatrix, true);
```

Comment:
I'm in no position to review the math on this. What would we be looking for, visually, to see if this is right? Would all shadows end up in the wrong place entirely, or could it be more subtle?